roles/satosa/tasks/main.yml

```diff
@@ -75,6 +75,14 @@
         state: latest
       when: satosa_microservices_git.changed
 
+    - name: Update all dependencies
+      shell: "bin/pip list --outdated --format=freeze | \
+              grep -v '^\\-e' | \
+              cut -d = -f 1 | \
+              xargs -rn1 bin/pip install -U"
+      args:
+        chdir: "{{ satosa_env_dir }}"
+
```

Comment:
I'm not sure that this is the proper solution: won't this update _all_ dependencies on every deploy?
I'm particularly afraid of the scenario in which a dependency would be updates between the moment of a deploy to test and the deploy to pilot, possible breaking things.

That having said, the current situation, ion which we _never_ update, clearly also isn't optimal.  Maybe we should set dependencies to a fixed version (but then, how would we know when to update one), or we should deploy complete (fixed) virtualenvs instead of creating the virtualenv during deploy (but then again, why not move to full containers then?).

Reply:
Good question